DatadogRUM/Sources/RUMMonitor/Monitor.swift

```diff
@@ -135,38 +133,34 @@ internal class Monitor: RUMCommandSubscriber {
     func process(command: RUMCommand) {
         // process command in event context
         featureScope.eventWriteContext { context, writer in
-            self.queue.sync {
-                let transformedCommand = self.transform(command: command)
+            let transformedCommand = self.transform(command: command)
 
-                _ = self.scopes.process(command: transformedCommand, context: context, writer: writer)
+            _ = self.scopes.process(command: transformedCommand, context: context, writer: writer)
 
-                if let debugging = self.debugging {
-                    debugging.debug(applicationScope: self.scopes)
-                }
+            if let debugging = self.debugging {
+                debugging.debug(applicationScope: self.scopes)
```

Comment:
Shouldn't we weakify self within this block?

Reply:
✅ done